src/Telemetry/Telemetry.cs

```diff
@@ -122,6 +122,26 @@ private void InitializeTelemetry(string productVersion, IConfiguration config)
             }
         }
 
+        public void TrackSQLClientEvent(string eventName, IDictionary<string, string> properties = null)
+        {
+            try
+            {
+                if (!this._initialized || !this.Enabled || this._client is null)
+                {
+                    return;
+                }
+                this._logger.LogTrace($"Sending event {eventName}");
+
+                this._client.TrackEvent($"{EventsNamespace}/{eventName}", properties, null);
```

Comment:
You should use TrackEvent here (see other Track* methods)

Reply:
yes but the TrackEvent accepts *TelemetryEventName* and IDictionary<*TelemetryPropertyName*, string> properties and the data we get from SqlClientListener cannot be predefined and sent - the reason why I used the TelemetryClient's TrackEvent.

Reply:
Oh wait, so first big question is why are you even sending this via telemetry? We shouldn't be doing that. We just want to log these events locally.